File: tech.stack.packages/ASP.NET Core/business/primarykey/__classes__PrimaryKey.cs
#header()
#set( $appName = $aib.getApplicationNameFormatted() )
#set( $className = $classObject.getName() )
using System;
using System.Collections;

namespace ${appName}.PrimaryKeys
{
    /// <summary>
    /// ${className} PrimaryKey class, encapsulating the key field(s) of an associated ${className} model 
    /// 
    /// @author    $aib.getAuthor()
    /// </summary>
	public class ${className}PrimaryKey
#set( $pkParentName = "" )
#if ( $classObject.hasParent() == true )
#set( $pkParentName = "${classObject.getParentName()}PrimaryKey" )
#end ##$classObject.hasParent() == true
#if ( $pkParentName.length() > 0 )
	: $pkParentName
#else
    : BasePrimaryKey
#end ##$pkParentName.length() > 0	
	{

//************************************************************************
// Public Methods
//************************************************************************
    	
        /// <summary>
        /// default constructor
        /// </summary>
		public ${className}PrimaryKey()
		{
		}
    
        /// <summary>
        /// Constructor with all arguments relating to the primary key
        /// </summary>
	    public ${classObject.getName()}PrimaryKey(    
#getAllPrimaryKeyArgumentsForDotNet( $classObject true )
		)
	    {
#set( $keyFields = $classObject.getAllPrimaryKeysInHierarchy() )
#foreach( $attribute in $keyFields )
##only provide accessors for primary-key fields
#if ( $attribute.isPrimaryKey() == true )
			${Utils.capitalizeFirstLetter( ${attribute.getAsArgument()} )} = ${attribute.getAsArgument()};
#end ##$attribute.isPrimaryKey() == true
#end ##foreach $attribute in $keyFields
    	}   

//************************************************************************
// Overload 
//************************************************************************

        /// <summary>
        /// Returns the key or keys associated with a associated ${className} model
        /// <returns></returns>
        /// </summary>
	    override public ArrayList keys()
	    {
			// assign the attributes to the Collection back to the parent
	        ArrayList keys = new ArrayList();
	        
#foreach( $primaryKey in $classObject.getAllPrimaryKeysInHierarchy() )
		    keys.Add( ${Utils.capitalizeFirstLetter( ${primaryKey.getAsArgument()} )} );
#end ##foreach $primaryKey in $classObject.getAllPrimaryKeysInHierarchy()

    	    return( keys );
	    }

#if ( $classObject.hasParent() == false )
        /// <summary>
        /// Returns the first assigned key
        /// <returns></returns>
        /// </summary>
		override public Object getFirstKey()
		{				   
#set( $pk = "$classObject.getAllPrimaryKeysInHierarchy().get(0)" )		
			return( ${Utils.capitalizeFirstLetter( $pk )} );
		}
#end
 
    
//************************************************************************
// Attributes
//************************************************************************

#getKeyFieldDeclarationsForDotNet()

	}
}

File: tech.stack.packages/AWS Lambda MongoDB/business/primarykey/__classes__PrimaryKey.java
#header()
package ${aib.getRootPackageName(true)}.#getPrimaryKeyPackageName();

import java.util.*;

/**
 * ${classObject.getName()} PrimaryKey class.
 * 
 * @author    $aib.getAuthor()
 */
// AIB : \#getPrimaryKeyClassDecl() 
#getPrimaryKeyClassDecl()
// ~AIB

//************************************************************************
// Public Methods
//************************************************************************

    /** 
     * default constructor - should be normally used for dynamic instantiation
     */
    public ${classObject.getName()}PrimaryKey() 
    {
    }    
    
    /** 
     * Constructor with all arguments relating to the primary key
     * 	        
// AIB : \#getAllAttributeJavaComments( true true )
#getAllAttributeJavaComments( true true )
// ~AIB     
     * @exception IllegalArgumentException
     */
    public ${classObject.getName()}PrimaryKey(    
// AIB : \#getAllPrimaryKeyArguments( \$classObject false )
#getAllPrimaryKeyArguments( $classObject false )
// ~AIB
                         ) 
    throws IllegalArgumentException
    {
        super();
// AIB : \#getKeyFieldAssignments()
#getKeyFieldAssignments()
// ~AIB 
    }   

    
//************************************************************************
// Access Methods
//************************************************************************

// AIB : \#getKeyFieldAccessMethods()
#getKeyFieldAccessMethods()
// ~AIB 	         	     

    /**
     * Retrieves the value(s) as a single List
     * @return List
     */
    public List keys()
    {
		// assign the attributes to the Collection back to the parent
        ArrayList keys = new ArrayList();
        
#foreach( $primaryKey in $classObject.getAllPrimaryKeysInHierarchy() )
		keys.add( $primaryKey.getAsArgument() );
#end## foreach $primaryKey in $classObject.getAllPrimaryKeysInHierarchy()

        return( keys );
    }

#if ( $classObject.hasParent() == false )
	public Object getFirstKey()
	{
		return( ${classObject.getAllPrimaryKeysInHierarchy().get(0)} );
	}
#end
#*
	/**
	 * Retrieves the values as a Collection of com.realmethods.foundational.common.parameters.Parameter parameters, 
	 * either as input or output parameters
	 * 
	 * @param  	asInput	indicates if the Parameters are input or output
	 * @return 	Collection of Parameters 
	 * @see		com.realmethods.foundational.common.parameter.Parameter
	 */
	public List getValuesAsParameters( boolean asInput )
	{
		ArrayList args = new ArrayList();

#foreach( $primaryKey in $classObject.getAllPrimaryKeysInHierarchy() )
#set( $type = $primaryKey.getRootType() )
		args.add( new ${type}Parameter( ${primaryKey.getAsArgument()}, asInput ) );
#end
						
		return args;
	}
*#
 
//************************************************************************
// Protected / Private Methods
//************************************************************************

    
//************************************************************************
// Attributes
//************************************************************************

 	

// DO NOT ASSIGN VALUES DIRECTLY TO THE FOLLOWING ATTRIBUTES.  SET THE VALUES
// WITHIN THE ${classObject.getName()} class.

// AIB : \#getKeyFieldDeclarations()
#getKeyFieldDeclarations()
// ~AIB 	        

}
